### PR TITLE
feat: allow ServiceAccountJwtAccessCredentials to sign scopes

### PR DESCRIPTION
**Pull request by bshaffer**
See https://google.aip.dev/auth/4111#scope-vs-audience

cc @shinfan

### CODE REVIEW

**Review by shinfan — APPROVED** [review]

File: src/Credentials/ServiceAccountJwtAccessCredentials.php [hunk 1]
```diff
@@ -128,10 +132,17 @@ public function updateMetadata(
...
     {
         $audience = $this->auth->getAudience();
-        if (empty($audience)) {
+        $scope = $this->auth->getScope();
+        if (empty($audience) && empty($scope)) {
             return null;
         }
 
+        if (!empty($audience) && !empty($scope)) {
+            throw new UnexpectedValueException(
```
Comment (dwsupplee) [inline src/Credentials/ServiceAccountJwtAccessCredentials.php:141]:
```suggestion
            throw new \UnexpectedValueException(
```

Would we be able to document this being thrown? Wondering if we need a test for this as well.

File: src/OAuth2.php [hunk 2]
```diff
@@ -437,9 +436,18 @@ public function toJwt(array $config = [])
...
+        if (!(is_null($this->getAudience()))) {
+            $assertion['aud'] = $this->getAudience();
+        }
+
         if (!(is_null($this->getScope()))) {
             $assertion['scope'] = $this->getScope();
         }
+
+        if (empty($assertion['scope']) && empty($assertion['aud'])) {
+            throw new \DomainException('one of scope or aud should not be null');
```
Comment (dwsupplee) [inline src/OAuth2.php:448]: Do we need a test to exercise this?

**Review by dwsupplee — CHANGES_REQUESTED** [review]

File: src/OAuth2.php (hunk 2 above)
Comment (bshaffer) [reply src/OAuth2.php:448]: good point, added!

File: src/Credentials/ServiceAccountJwtAccessCredentials.php (hunk 1 above)
Comment (bshaffer) [reply src/Credentials/ServiceAccountJwtAccessCredentials.php]: Good catch and yes, I added a test!

I'm happy to add `@throws`. In this library in general (not that it's the right thing to do), we haven't been documenting thrown exceptions.

Comment (dwsupplee) [reply src/Credentials/ServiceAccountJwtAccessCredentials.php]: I trust your judgement 👍

**Review by dwsupplee — APPROVED** [review]

**Pull request merged**